include/seqan3/range/view/take.hpp

```diff
@@ -457,7 +457,8 @@ class take_fn : public pipable_adaptor_base<take_fn<exactly, or_throw>>
      */
     template <std::ranges::ViewableRange urng_t>
     //!\cond
-        requires std::ranges::RandomAccessRange<urng_t> && std::ranges::SizedRange<urng_t>
+        requires std::ranges::RandomAccessRange<urng_t> && std::ranges::SizedRange<urng_t> &&
+                 std::is_lvalue_reference_v<urng_t>
```

Comment:
So lvalue views are fine too?
If this is not used for overload resolution we decided in one of the strategy meetings that it should be done as a static assert :point_up:

Reply:
From what I'm understanding, this is indeed for overload resolution... Because we want viewable ranges which are RandomAccess and Sized but not lvalue to go through the call above this one. @rrahn is that right?

Reply:
That is correct. We cannot take the subrange over a temporary view. So in this case we need to resolve to the most generic case which is returning a take_view.